### ISSUE TITLE
_send_signed_request() nonce bad error checking, might lead to generate empty certificates

### ISSUE THREAD

**Issue by zerodeux** [acmesh-official/acme.sh#5208]
This happened to me twice recently, it starts with those errors in a single row while renewing a single domain :
```
Please refer to https://curl.haxx.se/libcurl/c/libcurl-errors.html for error code: 35
Please refer to https://curl.haxx.se/libcurl/c/libcurl-errors.html for error code: 35
Please refer to https://curl.haxx.se/libcurl/c/libcurl-errors.html for error code: 35
Please refer to https://curl.haxx.se/libcurl/c/libcurl-errors.html for error code: 35
Sign failed: 
[...]
```

In my case I ended with a `<domain>.cer` file with a single byte (a newline). Acme.sh should have kept in intact, waiting for another later renewal to succeed.

I guess the problem is in `_send_signed_request()` which tries several ways of fetching a nonce, but does not make the proper test when they all failed. I propose this patch :
```
@@ -2167,7 +2167,7 @@
         _CACHED_NONCE="$(echo "$_headers" | grep -i "Replay-Nonce:" | _head_n 1 | tr -d "\r\n " | cut -d ':' -f 2)"
       fi
       _debug2 _CACHED_NONCE "$_CACHED_NONCE"
-      if [ "$?" != "0" ]; then
+      if [ -z "$_CACHED_NONCE" ]; then
         _err "Can not connect to $nonceurl to get nonce."
         return 1
       fi
```

**Comment by zerodeux** [comment]
I'm on acme.sh 3.0.6 and the latest is 3.0.7 : I checked the diff and I know the problem I report is not fixed in 3.0.7/latest.

**Comment by zerodeux** [comment]
This bug is rare and very difficult to reproduce, I won't be able to produce a "--debug 2" output.

The bug analysis is rather straigthforward, and so is the patch.
